src/app/location-map/location-map.component.ts

```diff
@@ -7,47 +7,54 @@ import leaflet from 'leaflet';
     styleUrls: ['./location-map.component.css'],
 })
 export class LocationMapComponent implements AfterViewInit {
-    @Output() getMapCoords = new EventEmitter();
-    @Output() getMap = new EventEmitter();
-    @Input() movableMarker: boolean;
-    @Input() coordsSet = [];
-    map;
-    startingLocation = [37.97404469468311, 23.71933726268805];
-    mapMarker;
-    constructor() {}
+    @Input() lat: number;
+    @Input() lng: number;
+    @Output() latChange = new EventEmitter<number>();
+    @Output() lngChange = new EventEmitter<number>();
+    @Input() editable: boolean;
 
-    ngAfterViewInit() {
-        this.initMap();
+    @ViewChild('mapEl') mapEl;
+
+    map: leaflet.Map;
+    latLngDefault: leaflet.LatLngTuple = [37.97404469468311, 23.71933726268805];
+    mapMarker: leaflet.Marker;
+
+    constructor() {
     }
 
-    private initMap() {
+    ngAfterViewInit() {
         leaflet.Icon.Default.prototype.options.shadowUrl = 'leaflet/marker-shadow.png';
-        this.map = leaflet.map('map').setView(this.startingLocation, 12);
-        this.mapMarker = leaflet.marker(this.startingLocation).addTo(this.map);
-        if (!(this.coordsSet.length === 0)) {
-            this.map.removeLayer(this.mapMarker);
-            this.map.setView(this.coordsSet, 12);
-            this.mapMarker = leaflet.marker(this.coordsSet).addTo(this.map);
-        } else {
-            this.getMapCoords.emit(this.startingLocation);
```

Comment:
Look at your previous code. You set a default marker location, put it on the map, then see if there are coordinates to set, and then set the marker another time. Not just unnecessary, but hard to read and could cause a map flickering.